### PR TITLE
SMTP configuration

### PR DESCRIPTION
**Pull request by jdartigalongue**

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1053#" title="Author self-approved">jdartigalongue</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.11.x/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@jdartigalongue: once the present PR merges, I will cherry-pick it on top of master in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/1053#issuecomment-549512514):

>/cherrypick master


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: pkg/util/baseyamlutil.go [hunk 1]
```diff
@@ -47,30 +47,25 @@ func ConvertYamlFileToRuntimeObjects(stringContent string) map[string]runtime.Ob
...
-			//log.V(1).Info("Got empty", "here", singleYaml)
 			continue
 		}
 
 		decode := scheme.Codecs.UniversalDeserializer().Decode
 		runtimeObject, groupVersionKind, err := decode([]byte(singleYaml), nil, nil)
 		if err != nil {
-			//log.V(1).Info("unable to decode a single yaml object, skipping", "singleYaml", singleYaml, "error", err)
-			continue
+			return nil, err
```
Comment (mphammer) [inline pkg/util/baseyamlutil.go]: Why does this check return an error but the checks before and after this one do `continue`?

Comment (mphammer) [reply pkg/util/baseyamlutil.go]: I get that this check is a little more important than the next one :) but maybe the next one should also error or log an Error "Unable to create ID for 'kind' due to 'err'"

File: pkg/polaris/types.go [hunk 2]
```diff
@@ -130,13 +130,25 @@ type MongoDBDetails struct {
 	Storage Storage `json:"storage,omitempty"`
 }
 
+type SMTPTLSMode string
+
+const (
+	SMTPTLSModeDisable         SMTPTLSMode = "disable"
+	SMTPTLSModeTryStartTLS     SMTPTLSMode = "try-starttls"
+	SMTPTLSModeRequireStartTLS SMTPTLSMode = "require-starttls"
+	SMTPTLSModeRequireTLS      SMTPTLSMode = "require-tls	"
```
Comment (mphammer) [inline pkg/polaris/types.go]: There is a weird space here

File: pkg/util/baseyamlutil.go (hunk 1 above)
Comment (jdartigalongue) [reply pkg/util/baseyamlutil.go]: Fixed. We'll just return the error.

File: pkg/polaris/types.go (hunk 2 above)
Comment (jdartigalongue) [reply pkg/polaris/types.go]: Good catch.

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1053#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/1053#pullrequestreview-311351045" title="LGTM">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.11.x/OWNERS)~~ [jdartigalongue,mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@jdartigalongue: new pull request created: #1054

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/1053#issuecomment-549512514):

>/cherrypick master


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**